tests/sklearn/test_sklearn_models.py

```diff
@@ -1763,3 +1796,39 @@ def test_linear_models_have_no_tlu(
 
     # Check that no TLUs are found within the MLIR
     check_circuit_has_no_tlu(fhe_circuit)
+
+
+# Test only tree-based models
+@pytest.mark.parametrize("model_class, parameters", get_sklearn_tree_models_and_datasets())
+@pytest.mark.parametrize("n_bits", [2, 3, 4, 5])
+def test_rounding_consistency(
+    model_class,
+    parameters,
+    n_bits,
+    load_data,
+    check_r2_score,
+    check_accuracy,
+    is_weekly_option,
+    verbose=True,
+):
+    """Test that Concrete ML witout and with rounding are 'equivalent'."""
+
+    if verbose:
+        print("Run check_rounding_consistency")
+
+    model, x = preamble(model_class, parameters, n_bits, load_data, is_weekly_option)
+
+    check_rounding_consistency(
+        model,
+        x,
+        predict_method=model.predict,
+        metric=check_accuracy,
```

Comment:
I don't think we should use `check_accuracy` for regressors 

ultimately, I think we should only check out `predict` for regressors and `predict_proba` for classifiers, as we usually do with such tests

Reply:
For regressors, I checked only predict method
For classifiers, I checked the predict and predict_proba methods.